File: leetcode2/1easy/최원준/Q645.java
package Leetcode.최원준;

/*
1. 아이디어 :


2. 시간복잡도 :
O( )

3. 자료구조/알고리즘 :

 */

import java.util.HashSet;
import java.util.Set;

public class Q645 {
    class Solution {
        public int[] findErrorNums(int[] nums) {
            int n = nums.length;
            int total = n * (n+1) / 2;
            int sum = 0;

            int[] ans = new int[2];
            Set<Integer> dup = new HashSet<>();

            for (int num : nums) {
                sum+=num;
                if (dup.contains(num)) ans[0] = num;
                dup.add(num);
            }
            ans[1] = ans[0] + total - sum;


            return ans;
        }
    }
}

File: leetcode2/1easy/강병호/Q645.py
class Leetcode.이재훈.이재훈.Solution:
    def findErrorNums(self, nums: List[int]) -> List[int]:
        duplicate, miss  = -1, -1

        for num in nums:
            idx = abs(num) - 1
            if nums[idx] < 0:
                duplicate = abs(num)
            else:
                nums[idx] *= -1

        for i, num in enumerate(nums):
            if num > 0:
                miss = i + 1
                break

        return [duplicate, miss]